service/docs/source/geopm_agent_cpu_activity.7.rst

```diff
@@ -0,0 +1,256 @@
+are policy values.
+Setting ``Fe`` & ``Fmax`` for a domain to the same value can
+be used to force the entire application to run at a fixed frequency.
+
+  ``CPU_FREQ_MAX``\ :
+      The maximum cpu frequency in hertz that the algorithm is
+      allowed to choose.  If NAN is passed, it will use the
+      maximum available frequency by default.
+
+  ``CPU_FREQ_EFFICIENT``\ :
+      The minimum cpu frequency in hertz that the algorithm is
+      allowed to choose.  If NAN is passed, it will use the system
+      minimum frequency by default.
+
+  ``CPU_UNCORE_FREQ_MAX``\ :
+      The maximum cpu uncore frequency in hertz that the algorithm is
+      allowed to choose.  If NAN is passed, it will use the
+      maximum available frequency by default.
+
+  ``CPU_UNCORE_FREQ_EFFICIENT``\ :
+      The minimum cpu uncore frequency in hertz that the algorithm is
+      allowed to choose.  If NAN is passed, it will use the system
+      minimum frequency by default.
+
+  ``CPU_PHI``\ :
+      The performance bias knob.  The value must be between
+      0.0 and 1.0. If NAN is passed, it will use 0.5 by default.
+
+  ``CPU_UNCORE_FREQ_#``\ :
+      The uncore frequency associated with the same numbered
+      maximum memory bandwidth.
+      Used to build a mapping of uncore frequencies to maximum
+      memory bandwidths for frequency steering.
+
+  ``MAX_MEMORY_BANDWIDTH_#``\ :
+      The maximum possible memory bandwidth associated with the
+      same numbered uncore frequency.
+      Used to build a mapping of uncore frequencies to maximum
+      memory bandwidths for frequency steering.
+
+Policy Generation
+-----------------
+
+Generating a policy for the CPU Compute Activity agent requires system
+characterization using the GEOPM experiment infrastructure.  This can be
+accomplished by running::
+
+    gen_slurm.sh 1 arithmetic_intensity_benchmark uncore_frequency_sweep
+
+The generated test.sbatch should be modified to enable Memory Bandwidth
+Monitoring using the following::
+
+    srun -N ${SLURM_NNODES} geopmwrite MSR::PQR_ASSOC:RMID board 0 0
+    srun -N ${SLURM_NNODES} geopmwrite MSR::QM_EVTSEL:RMID board 0 0
+    srun -N ${SLURM_NNODES} geopmwrite MSR::QM_EVTSEL:EVENT_ID board 0 2
+
+Without this enabling the uncore bandwidth characteriztaion analysis scripts
+will not be able to accurately determine the maximum memory bandwidth at each
+uncore frequency.
+
+Additionally the test.sbatch should be modified to include the following
+experiment options, where ``MAX_UNCORE_FREQ``, ``MIN_UNCORE_FREQ``,
+``MAX_CORE_FREQ``, and ``MIN_CORE_FREQ`` are replaced with the relevant
+system (or administrator chosen) values::
+
+    --geopm-report-signals="MSR::QM_CTR_SCALED_RATE@package,CPU_UNCORE_FREQUENCY_STATUS@package,MSR::CPU_SCALABILITY_RATIO@package,CPU_FREQUENCY_MAX_CONTROL@package,CPU_UNCORE_FREQUENCY_MIN_CONTROL@package,CPU_UNCORE_FREQUENCY_MAX_CONTROL@package" \
+    --min-frequency=MIN_CORE_FREQ \
```

Comment:
Suggestion (instead of using MIN_CORE_FREQ, ...): --min-frequency=<min system frequency> \